pkg/cmd/roachtest/tests/sysbench.go

```diff
@@ -179,3 +195,89 @@ func registerSysbench(r registry.Registry) {
 		})
 	}
 }
+
+type sysbenchMetrics struct {
+	Time         int64  `json:"time"`
+	Threads      string `json:"threads"`
+	Transactions string `json:"transactions"`
+	Qps          string `json:"qps"`
+	ReadQps      string `json:"readQps"`
+	WriteQps     string `json:"writeQps"`
+	OtherQps     string `json:"otherQps"`
+	P95Latency   string `json:"p95Latency"`
+	Errors       string `json:"errors"`
+	Reconnects   string `json:"reconnects"`
+}
+
+// exportSysbenchResults parses the output of `sysbench` into a JSON file
+// and writes it to the perf directory that roachperf expects. Sysbench does
+// have a way to customize the report output via injecting a custom
+// `sysbench.hooks.report_intermediate` hook, but then we would lose the
+// human-readable output in the test itself.
+func exportSysbenchResults(t test.Test, result string, start time.Time) error {
+	// Parse the results into a JSON file that roachperf understands.
+	// The output of the results look like:
+	// 		1. Start up information.
+	//		2. Benchmark metrics timeseries every second.
+	//		3. Benchmark metrics summary.
+	//
+	// For roachperf, we care about 2, so filter out any line that
+	// doesn't start with a timestamp.
+	//
+	// An example line of this output:
+	// [ 1s ] thds: 256 tps: 2696.16 qps: 57806.17 (r/w/o: 40988.38/11147.98/5669.82) lat (ms,95%): 196.89 err/s: 21.96 reconn/s: 0.00
+
+	filter := "\\[ [\\d]+s \\].*"
+	regex, err := regexp.Compile(filter)
+	if err != nil {
+		return err
+	}
+
+	var metricBytes []byte
+
+	s := bufio.NewScanner(strings.NewReader(result))
+	for s.Scan() {
+		if matched := regex.MatchString(s.Text()); !matched {
+			continue
```

Comment:
Should this be an error, or are there lines we don't expect to match? If latter, maybe we should count the number of parsed lines and log it, for ease of debugging; i.e., thinking about the case when all of a sudden we get an empty output, say due to a format change upstream.